### PR TITLE
Added support for infinite scrolling

### LINKED ISSUE #62: Endless scrolling/ looping around the applications
**Is your feature request related to a problem? Please describe.**
Sometimes you go in the wrong direction and then have to turn around and go back

**Describe the solution you'd like**
endless scrolling in the Application selection menu. (if you keep scrolling right, you go through all applications and end up at your starting point)

### PR DESCRIPTION
**Pull request by SchemingWeasels**
## Issues
 - _Attempt to Resolve_ #62 
 - Continued from #65 

## Description
Added support for infinite scrolling on the embedded side.
Requires support on the desktop app.

## Types of changes
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation Update (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

- [X] Requested changes are in a branch
- [X] Compiled and tested requested changes on target hardware (PC, device)
- [ ] Updated the documentation, if necessary
- [ ] Updated the LICENSES file, if necessary
- [X] Reviewed the [guidelines for contributing](../CONTRIBUTING.md) to this repository

## Further comments

Another element has been added to the settings struct (default true) for infinite scrolling. I have added it as the 4th index of the settings update packageBuffer.

The feature works as is but there is no way to turn it off without updating the desktop app or compiling your own Arduino bin.

I don't really know how to work on the desktop app. Maybe a point in the right direction of where I should start would be great @t3knomanzer .

### CODE REVIEW

**Comment by t3knomanzer** [comment]
Thank you for creating a branch!
Could you squash the commits so that we have a cleaner history when merging?

[Here is an example](https://medium.com/@slamflipstrom/a-beginners-guide-to-squashing-commits-with-git-rebase-8185cf6e62ec) if you don't know how to do it.

File: Embedded/MaxMix/Display.ino [hunk 1]
```diff
@@ -35,23 +35,41 @@ void DisplayAppNavigateScreen(Adafruit_SSD1306* display, Item* item, int8_t item
 {
   display->clearDisplay();
 
-  // Left Arrow
-  if(itemIndex > 0)
-  {
+
+  if(settings.continuousScroll && itemCount > 0)
```
Comment (t3knomanzer) [inline Embedded/MaxMix/Display.ino]: We are trying to not use global variables in functions outside of the Maxmix.ino file.  
Could you add a new argument to this function "uint8_t continuousScroll" and remove the reference to settings here?

Also if there is only 1 item, the arrows should be hidden. Therefore you may want to have 2 if statements like so:
`if(continuousScroll)
    if(itemCount > 1)
    ...`

**Review by t3knomanzer — CHANGES_REQUESTED** [review]

File: Embedded/MaxMix/Display.ino (hunk 1 above)
Comment (SchemingWeasels) [reply Embedded/MaxMix/Display.ino]: Alright. Do I squash these with the initial commits or do I leave them a separate commit?

Comment (t3knomanzer) [reply Embedded/MaxMix/Display.ino]: Yeah leve it separate, it’s ok. 
Thanks a lot for the contribution, it’s great!

**Comment by t3knomanzer** [comment]
I’ve submitted the implementation for desktop app side. 
After the changes of the review are implemented it should be good to go @SchemingWeasels

**Comment by SchemingWeasels** [comment]
I have figured out why the rebasing wasn't working. Fixed the horrid mess of commits. Also, I should have named the branch something more informative. A lil late now tho.

Thanks for the help @t3knomanzer

**Review by t3knomanzer — APPROVED** [review]

**Pull request merged**